File: WOT-iOS/WOT-iOS/core/WOTRequests/instances/Tanks/Engines/WOTWEBRequestTankEngines.m
//
//  WOTWEBRequestTankEngines.m
//  WOT-iOS
//
//  Created by Pavel Yeshchyk on 6/18/15.
//  Copyright (c) 2015 Pavel Yeshchyk. All rights reserved.
//

#import "WOTWEBRequestTankEngines.h"

@implementation WOTWEBRequestTankEngines

- (NSDictionary *)query {
    
    NSMutableDictionary *result = [[NSMutableDictionary alloc] init];
    result[WOT_KEY_APPLICATION_ID] = [NSString valueOrSpaceString:self.applicationID];
    result[WOT_KEY_FIELDS] = [NSString valueOrSpaceString:self.args[WOT_KEY_FIELDS]];
    return result;
}

- (NSString *)path {
    
    return @"wot/encyclopedia/tankengines/";
}

@end

File: WOT-iOS/WOTData/WOTData/WebServices/requests/WOTWEBRequestTankEngines.swift
//
//  WOTWEBRequestTankEngines.swift
//  WOTData
//
//  Created by Pavel Yeshchyk on 1/14/20.
//  Copyright © 2020 Pavel Yeshchyk. All rights reserved.
//

import Foundation

@objc
public class WOTWEBRequestTankEngines: WOTWEBRequest, WOTModelServiceProtocol {
    
    override public var method: String { return "POST" }

    @available(*, deprecated, message: "TO be refactored")
    @objc
    public static func modelClassName() -> String {
 //        return NSStringFromClass(TanksEngines.self)
        return ""
    }

    @available(*, deprecated, message: "TO be refactored")
    @objc
    public func instanceModelClass() -> AnyClass? {
        return NSClassFromString( type(of: self).modelClassName() )
    }

    override public var path: String {
        return "/wot/encyclopedia/tankengines/"
    }
}
